### ISSUE TITLE
Test failure: test_ascconv_parse

### ISSUE THREAD

**Issue by sanjayankur31** [nipy/nibabel#973]
Hello,

I'm seeing a test failure while building nibabel-3.2.0 on Fedora.

```
=================================== FAILURES ===================================
______________________________ test_ascconv_parse ______________________________

    def test_ascconv_parse():
        with open(ASCCONV_INPUT, 'rt') as fobj:
            contents = fobj.read()
>       ascconv_dict, attrs = ascconv.parse_ascconv(contents, str_delim='""')

nibabel/nicom/tests/test_ascconv.py:20:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
nibabel/nicom/ascconv.py:214: in parse_ascconv
    atoms = assign2atoms(assign)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

assign_ast = <ast.Assign object at 0x7f2a23b1faf0>
default_class = <class 'int'>

    def assign2atoms(assign_ast, default_class=int):
        """ Parse single assignment ast from ascconv line into atoms

        Parameters
        ----------
        assign_ast : assignment statement ast
            ast derived from single line of ascconv file.
        default_class : class, optional
            Class that will create an object where we cannot yet know the object
            type in the assignment.

        Returns
        -------
        atoms : list
            List of :class:`atoms`.  See docstring for :class:`atoms`.  Defines
            left to right sequence of assignment in `line_ast`.
        """
        if not len(assign_ast.targets) == 1:
            raise AscconvParseError('Too many targets in assign')
        target = assign_ast.targets[0]
        atoms = []
        prev_target_type = default_class  # Placeholder for any scalar value
        while True:
            if isinstance(target, ast.Name):
                atoms.append(Atom(target, prev_target_type, target.id))
                break
            if isinstance(target, ast.Attribute):
                atoms.append(Atom(target, prev_target_type, target.attr))
                target = target.value
                prev_target_type = OrderedDict
            elif isinstance(target, ast.Subscript):
>               index = target.slice.value.n
E               AttributeError: 'int' object has no attribute 'n'

nibabel/nicom/ascconv.py:92: AttributeError
=============================== warnings summary ===============================
nibabel/externals/tests/test_netcdf.py::test_itemset_no_segfault_on_readonly
nibabel/tests/test_proxy_api.py::TestMinc1API::test_array_interface_with_dtype
nibabel/tests/test_proxy_api.py::TestMinc1API::test_asarray
nibabel/tests/test_proxy_api.py::TestMinc1API::test_fileobj_isolated
nibabel/tests/test_proxy_api.py::TestMinc1API::test_header_isolated
nibabel/tests/test_proxy_api.py::TestMinc1API::test_is_proxy
nibabel/tests/test_proxy_api.py::TestMinc1API::test_ndim
nibabel/tests/test_proxy_api.py::TestMinc1API::test_proxy_slicing
nibabel/tests/test_proxy_api.py::TestMinc1API::test_shape
  /builddir/build/BUILD/nibabel-3.2.0/nibabel/externals/netcdf.py:320: RuntimeWarning: Cannot close a netcdf_file opened with mmap=True, when netcdf_variables or arrays referring to its data still exist. All data arrays obtained from such files refer directly to data on disk, and must be copied before the file can be cleanly closed. (See netcdf_file docstring for more information on mmap.)
    warnings.warn((

nibabel/streamlines/tests/test_tractogram.py::TestTractogram::test_creating_invalid_tractogram
nibabel/streamlines/tests/test_tractogram.py::TestTractogram::test_tractogram_creation
  /usr/lib64/python3.9/site-packages/numpy/core/_asarray.py:83: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
    return array(a, dtype, copy=False, order=order)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED nibabel/nicom/tests/test_ascconv.py::test_ascconv_parse - AttributeErr...
= 1 failed, 4698 passed, 110 skipped, 6 xfailed, 11 warnings in 125.12s (0:02:05) =
```

This is with python 3.9,  scipy-1.5.4, numpy-1.19.4.

**Comment by effigies** [comment]
Yes, fixed in #963. I can release a 3.2.1 to make sure 3.9 is supported in a released version.

**Comment by sanjayankur31** [comment]
That'll be great. Thanks very much @effigies
